### PR TITLE
Sanity check for sparsity pattern

### PR DESCRIPTION
**Pull request by AnHeuermann**
### Related Issues

#8459 

### Purpose

Workaround for fishy sparsity patterns.
Also resolving memory leak in initializaton.c when allocating static data a second time.

### CODE REVIEW

File: testsuite/simulation/modelica/equations/DAEexample.mo [hunk 1]
```diff
@@ -14,12 +14,3 @@ equation
   (1 + 0.5*sin(y))*der(x) + der(y) = a*sin(time);
   x-y = exp(-0.9*x)*cos(y);
 end DAEexample;
-
-// class DAEexample
```
Comment (AnHeuermann) [inline testsuite/simulation/modelica/equations/DAEexample.mo:18]: Why did we remove this?

**Review by AnHeuermann — COMMENTED** [review]
I think we are more or less ready to merge.

File: testsuite/simulation/modelica/equations/DAEexample.mo (hunk 1 above)
Comment (perost) [reply testsuite/simulation/modelica/equations/DAEexample.mo:18]: The file isn't used as a test case, it's just loaded by the corresponding mos-script. So if you remove the expected result you might as well remove the test header from the top of the file too to make it clear it's not a test case, or even just remove the file completely and load the model with `loadString` instead.

Comment (phannebohm) [reply testsuite/simulation/modelica/equations/DAEexample.mo:18]: Whoops, Idk why I overlooked the header, I thought it was old code. I'll revert it and perhaps make a separate PR to remove the unnecessary separation into mo and mos.

**Pull request merged**